File: inflow-backend-main/src/main/java/com/john/inflow/service/UserService.java
package com.john.inflow.service;

import com.john.inflow.dto.request.CreateUserRequest;
import com.john.inflow.dto.request.UpdateUserRequest;
import com.john.inflow.dto.response.UserResponse;

import java.util.List;

public interface UserService {
    UserResponse create(CreateUserRequest request);
    UserResponse getById(Integer id);
    List<UserResponse> getAll();
    UserResponse update(Integer id, UpdateUserRequest request);
    void delete(Integer id);
}

File: src/services/user.service.ts
import api from "./api";
import {
  User,
  CreateUserRequest,
  UpdateUserRequest,
} from "@/types/user.types";
import {
  ApiResponse,
  PaginatedResponse,
  PaginationParams,
} from "@/types/api.types";

export const userService = {
  getAll: (params?: PaginationParams) =>
    api.get<PaginatedResponse<User>>("/users", { params }),

  getById: (id: number) =>
    api.get<ApiResponse<User>>(`/users/${id}`),

  create: (data: CreateUserRequest) =>
    api.post<ApiResponse<User>>("/users", data),

  update: (id: number, data: UpdateUserRequest) =>
    api.put<ApiResponse<User>>(`/users/${id}`, data),

  delete: (id: number) =>
    api.delete<ApiResponse<void>>(`/users/${id}`),
};
